header for new method to adjust the particle coordinates to match a n…

egstern opened the pull request:
…ew reference energy

Just the start of this feature. The plan is to allow the full phase to be used in acceleration, but save the correct lattice energy which will be restored at the beginning of the next turn end action.

egstern committed:
header for new method to adjust the particle coordinates to match a new reference energy

egstern committed:
forgot this file

egstern committed:
corrected definition of adjust_bunch_particle_reference_energy()

egstern committed:
attempted implementation of adjust reference momentum and coordinates but doesnt build

egstern committed:
changed AdjustCoords object initialization to aggregate initialization which fixes build failure

egstern committed:
Just a name change to the adjust_particles_...
A test for the method

Commit:
Merge branch 'devel3' into 265-egsternneed-accelerating-phase-separate-from-reference-energy

s-sajid-ali commented:
Merged `devel3` to get the fix from https://github.com/fnalacceleratormodeling/synergia2/pull/273

egstern committed:
Upstream fix for building on perlmutter

Merge branch 'devel3' into 265-egsternneed-accelerating-phase-separate-from-reference-energy

egstern commented:
The feature is limited to the Bunch::adjust_particles_reference_energy
